step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.automation.packages.execution;
+
+import step.automation.packages.AutomationPackage;
+import step.automation.packages.AutomationPackageManager;
+import step.core.accessors.AbstractOrganizableObject;
+import step.core.execution.ExecutionContext;
+import step.core.plans.Plan;
+import step.core.repositories.*;
+import step.functions.accessor.FunctionAccessor;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class IsolatedAutomationPackageRepository extends AbstractRepository {
+
+    public static final String REPOSITORY_PARAM_CONTEXTID = "contextid";
+
+    private final ConcurrentHashMap<String, AutomationPackageManager> inMemoryPackageManagers = new ConcurrentHashMap<>();
+
+    protected IsolatedAutomationPackageRepository() {
+        super(Set.of(REPOSITORY_PARAM_CONTEXTID));
+    }
+
+    @Override
+    public ArtefactInfo getArtefactInfo(Map<String, String> repositoryParameters) throws Exception {
+        AutomationPackage automationPackage = getAutomationPackageForContext(repositoryParameters);
+        if (automationPackage == null) {
+            return null;
+        }
+        ArtefactInfo info = new ArtefactInfo();
+        // TODO: type?
+        info.setType("automationPackage");
```

Comment:
The method getArtefactInf won't be used for the REST service /execute which will directly call importArtefact. It would be used only if we want to trigger an execution from the UI view #/root/repository. We might introduce a new type indeed